libmobilecoin/include/transaction.h

```diff
@@ -515,9 +537,202 @@ bool mc_memo_sender_with_payment_request_memo_get_payment_request_id(
 )
 MC_ATTRIBUTE_NONNULL(1, 2);
 
+/* ==== Gift Code Memo Builders ==== */
 
-/* ==== Decrypt Memo Payload ==== */
+/// # Preconditions
+///
+/// * `gift_code_funding_note` - must be a null-terminated C string containing
+/// up to 54 valid UTF-8 bytes. The actual note stored on chain is up to 53 null
+/// terminated UTF-8 bytes unless the note is exactly 53 utf-8 bytes long,
+/// in which case, no null bytes are stored. If the C string passed here is
+/// exactly 54 bytes, the last byte MUST be null and that byte will be
+/// removed prior to storage on chain.
+McTxOutMemoBuilder* MC_NULLABLE mc_memo_builder_gift_code_funding_create(
+        const char* MC_NONNULL gift_code_funding_note
+)
+MC_ATTRIBUTE_NONNULL(1);
 
+/// # Preconditions
+///
+/// * `gift_code_sender_note` - must be a null-terminated C string containing up
+/// to 58 valid UTF-8 bytes. The actual note stored on chain is up to 57 null
+/// terminated UTF-8 bytes unless the note is exactly 57 utf-8 bytes long,
+/// in which case, no null bytes are stored. If the C string passed here is
+/// exactly 58 bytes, the last byte MUST be null and that byte will be
+/// removed prior to storage on chain.
+McTxOutMemoBuilder* MC_NULLABLE mc_memo_builder_gift_code_sender_create(
+        const char* MC_NONNULL gift_code_sender_note
+)
+MC_ATTRIBUTE_NONNULL(1);
+
+/// # Preconditions
+///
+/// * `global_index` - must be the global TxOut index of the originally funded
+///   gift code TxOut
+McTxOutMemoBuilder* MC_NULLABLE mc_memo_builder_gift_code_sender_create(
+        uint64_t global_index
+);
+
+/* ==== GiftCodeFundingMemo ==== */
+
+/// # Preconditions
+///
+/// * `tx_out_public_key` - must be a valid 32-byte Ristretto-format scalar.
+/// * `fee` - must be an integer less than or equal to 2^56
+/// * `gift_code_funding_note` - must be a null-terminated C string containing
+/// up to 54 valid UTF-8 bytes. The actual note stored on chain is up to 53 null
+/// terminated UTF-8 bytes unless the note is exactly 53 utf-8 bytes long,
+/// in which case, no null bytes are stored. If the C string passed here is
+/// exactly 54 bytes, the last byte MUST be null and that byte will be
+/// removed prior to storage on chain.
+/// * `out_memo_data` - length must be >= 64.
+///
+/// # Errors
+///
+/// * `LibMcError::InvalidInput`
+bool mc_memo_gift_code_funding_memo_create(
```

Comment:
I'm not entirely sure WHY the arguments aren't documented anywhere, they could probably stand to be. I'm not entirely sure they'd reflect properly into swift, but if they were, it'd probably be useful. 

As far as when something returns `true` vs `false` - that's more or less because you can't panic/it's undefined behavior to panic across an ffi boundary and you need to return a value indicating that you didn't succeed". `bool` is the default if you aren't returning a value from the ffi.

Success/Error defaults are defined here:
https://github.com/mobilecoinfoundation/mobilecoin/blob/master/libmobilecoin/src/common/into_ffi.rs

And implementation of logging the error + returning a failed value is here:
https://github.com/mobilecoinfoundation/mobilecoin/blob/master/libmobilecoin/src/common/boundary.rs